pkg/tasks/prometheus.go

```diff
@@ -420,3 +418,13 @@ func (t *PrometheusTask) Run(ctx context.Context) error {
 
 	return nil
 }
+
+func (t *PrometheusTask) Validate(ctx context.Context) *client.ValidationError {
+	klog.V(4).Info("validate Prometheus object")
+	if t.prometheus == nil {
```

Comment:
this should never happen right? if so, I'd prefer a hard fail with panic()

Reply:
I wasn't too sure about `panic` :).  Will change to use `panic`